packages/server-wallet/src/wallet/index.ts

```diff
@@ -81,21 +81,23 @@ export class Wallet implements WalletInterface {
   }
 
   async createChannel(args: CreateChannelParams): SingleChannelResult {
+    const {participants, appDefinition, appData, allocations} = args;
+    const outcome: Outcome = deserializeAllocations(allocations);
+
+    const channelNonce = await Nonce.next(participants.map(p => p.signingAddress));
```

Comment:
Just so I understand: by fetching the nonce outside of the transaction, it could be that the transaction later will fail (e.g. if another thread successfully created a channel with that nonce in the meantime?).

Reply:
`Nonce.next` is guaranteed to return a unique nonce, for the same set of participants! Another thread that called `Nonce.next` with the same participants would get a different nonce, so there is no collision like you describe.